src/battle_script_commands.c

```diff
@@ -6027,9 +6027,9 @@ static void Cmd_moveend(void)
                     // Attacker is the damage-dealer, battler is mon to be switched out
                     if (IsBattlerAlive(battler)
                       && gBattlerAttacker != battler
-                      && GetBattlerHoldEffect(battler, TRUE) == HOLD_EFFECT_EJECT_BUTTON
                       && BATTLER_TURN_DAMAGED(battler)
-                      && CountUsablePartyMons(battler) > 0)  // Has mon to switch into
+                      && CountUsablePartyMons(battler) > 0  // Has mon to switch into
+                      && GetBattlerHoldEffect(battler, TRUE) == HOLD_EFFECT_EJECT_BUTTON)
```

Comment:
These actually take lots of cycle anyway(because of sorting), and this makes it even worse, because most of the time everything passes except the holdEffect call. Same for Red Card and Eject Pack